### ISSUE TITLE
Error message on UnboundLocalError: local variable 'action' referenced before assignment

### ISSUE THREAD

**Issue by jfarbowitz** [amiaopensource/grabbags#46]
```
Traceback (most recent call last):
  File "/Users/Jonathan/Documents/GitHub/grabbags/grabbags/gui.py", line 184, in _run
    worker.run(paths, self.context.options.get_settings())
  File "/Users/Jonathan/Documents/GitHub/grabbags/grabbags/gui.py", line 274, in run
    grabbags.main(args)
  File "/Users/Jonathan/Documents/GitHub/grabbags/grabbags/grabbags.py", line 305, in main
    {"count": len(successes), "action": action}
UnboundLocalError: local variable 'action' referenced before assignment
```

Henry says it might have to do with not having all situations covered in the Else elif statements. Might want to put arg parsing in a separate function.

**Comment by jfarbowitz** [comment]
Possible fix, in ln 270 edit to read:
`if args.validate or args.no_checksums`

**Comment by jfarbowitz** [comment]
Closed with commit 9e86b813f5f8f046e1d9ac8094eccf24fa87afcb

Henry created action_types
